.github/workflows/pull-request-ci.yaml

```diff
@@ -42,6 +42,7 @@ jobs:
       - name: Checkout KubeCF
         uses: actions/checkout@v2
         with:
+          ref: ${{ github.event.pull_request.head.sha }}
```

Comment:
I believe this is unnecessary; when triggered by the PR, it already defaults to the head of the PR.  And when manually triggered, this will just error out.  (This is why this job is triggered on labelling — so that it will have a correct `HEAD`; if we just did it in the scheduler it wouldn't be associated with the correct branch.)

The problem is that the lint job isn't using the correct commit, not the CI job.

Reply:
> And when manually triggered, this will just error out.

Well, that's definitely not good. But on my tests I didn't see that behaviour, tbh.

The intent was to always check the HEAD of the branch on CI and lint workflows. Yet later I removed ` ref: ${{ github.event.pull_request.head.sha }}`  from the lint workflow because of security reasons:
*one could open a PR changing the lint workflow to consume the GH secrets and print them, and it would get run automatically*.